app/controllers/users_controller.rb

```diff
@@ -22,12 +24,32 @@ def login
 
   def logout
     token = request.headers['Authorization'].split(' ')[1]
-    BlacklistToken.create(token: token)
+    BlacklistToken.create(token:)
     render json: { message: 'successfully logged out' }, status: :ok
   end
 
+  def destroy
+    unless @user.destroy
+      return render json: { message: @user.errors.full_messages },
+                    status: :unprocessable_entity
+    end
+
+    render json: { message: 'Your account deleted successfully' }, status: :ok
+  end
+
   private
 
+  def set_user
+    @user = User.find_by(id: params[:id])
+    return render json: { message: @user.errors.full_messages }, status: :not_found if @user.nil?
+  end
+
+  def validate_owner
+    return unless params[:id] == @current_user.id
+
+    render json: { message: 'Not authorized to delete this account' }, status: :unauthorized
+  end
+
```

Comment:
Do we need to take id from the params when we are deleting the "current_user"?

Since, the user is logged in and if they click on delete button, we know that the logged in user is trying to delete their account. So can't we just do `@current_user.destroy`?

Reply:
if we think in frontend then you are right, but destroy function in rails takes id in params, and their are chances of sending a api call of a user who is trying to send someone else's account id.

Reply:
override the destroy function not to take id in params. google it

Reply:
Create a custom delete action, let's call it `delete_current_user`. When user goes to route: `DELETE - /users/me`, it should execute the code written in `delete_current_user` and delete the current user.

Reply:
Remember, rails is just a tool. But the working logic has to be defined by you.